### PR TITLE
Adds function AI_CalcMoveScore to more easily control score increases

### PR DESCRIPTION
**Pull request by AlexOn1ine**
## Description

Often times scores in `AI_CheckViability` were increased uncontrollable through various functions, e.g. `IncreaseStatUpScore`. This isn't necessarily bad because some moves are really good and deserve a high score but it collides with other functions that control score because we won't exactly know by how much a score will be increased for a given move which makes it harder to adjust scores.

My possible solution with `AI_CalcMoveScore` is to calculate a temp score and return a final score based on the moves viability. This allows for more control because we will always know the threshold by which a score can be increased. We are also free to increase a score of an effect by any amount without worries to disrupt any other interaction.  

For now I simply created the function and moved the checks to it and haven't written any tests but I would be glad to write some if specific scenarios need testing.

Also in the future, if I get the okay, I would like to phase out `AI_CheckBadMove`. I think we can manage everything through score increases. Having a function that also decreases scores seems redundant.

### CODE REVIEW

**Review by mrgriffin — DISMISSED** [review]
I really like the idea, it should hopefully improve the AI's strength :)

I've approved this because I think it all looks good, but there is one suggestion about a comment that I think shouldn't have been changed that I'd like you to apply before merging please!

I've left a bunch of comments about the AI's logic that are all non-blocking because afaict none of them are about your changes, they're about how the AI worked before this PR. If there's any ideas you agree with then please raise an issue and we can look at getting them done in the future!

File: src/battle_ai_main.c [hunk 1]
```diff
@@ -4669,6 +4628,252 @@ static s32 AI_CheckViability(u32 battlerAtk, u32 battlerDef, u32 move, s32 score
...
+                        ADJUST_SCORE(3);
+                    break;
+                case MOVE_EFFECT_POISON:
+                    IncreasePoisonScore(battlerAtk, battlerDef, move, &score);
+                    break;
+                case MOVE_EFFECT_CLEAR_SMOG:
+                    score += AI_TryToClearStats(battlerAtk, battlerDef, FALSE);
+                    break;
+                case MOVE_EFFECT_SPECTRAL_THIEF:
+                    score += AI_ShouldCopyStatChanges(battlerAtk, battlerDef);
```
[the remark this answers is not in the document]
Comment (AlexOn1ine) [reply src/battle_ai_main.c:4736]: yep, agreed.

**Comment by AlexOn1ine** [comment]
Thanks for the helpful comments. I commented on a couple where it was immediately obvious to me. I'll go through the rest later. 
So the options are. Either merge the PR right now, and address the comments in a follow up PR or address them here. I'm fine with both.

**Review by mrgriffin — APPROVED** [review]

**Comment by mrgriffin** [comment]
I merged it because it's possible that the follow-up stuff might end up needing more discussion, whereas these bits are all uncontroversial (imo!) :)

**Pull request merged**